libs/aws/langchain_aws/llms/q_business.py

```diff
@@ -0,0 +1,185 @@
+    """Store the full response from Amazon Q."""
+
+    parent_message_id: Optional[str] = None
+    """AWS region name. If not provided, will be extracted from environment."""
+
+    conversation_id: Optional[str] = None
+    """AWS region name. If not provided, will be extracted from environment."""
+
+    chat_mode: str = "RETRIEVAL_MODE"
+    """AWS region name. If not provided, will be extracted from environment."""
+
+    credentials_profile_name: Optional[str] = None
+    """The name of the profile in the ~/.aws/credentials or ~/.aws/config files, which
+    has either access keys or role information specified.
+    If not specified, the default credential profile or, if on an EC2 instance,
+    credentials from IMDS will be used.
+    See: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
+    """
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "amazon_q"
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the Amazon Q client."""
+        super().__init__(**kwargs)
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> str:
+        """Call out to Amazon Q service.
+
+        Args:
+            prompt: The prompt to pass into the model.
+            stop: Optional list of stop words to use when generating.
+
+        Returns:
+            The string generated by the model.
+
+        Example:
+            .. code-block:: python
+
+                response = llm.invoke("Tell me a joke.")
+        """
+        try:
+            print("Prompt Length (Amazon Q ChatSync API takes a maximum of 7000 chars)")
+            print(len(prompt))
+
+            # Prepare the request
+            request = {
+                'applicationId': self.application_id,
+                'userMessage': prompt,
+                'chatMode':self.chat_mode,
+            }
+            if self.conversation_id: 
+                request.update({ 
+                'conversationId': self.conversation_id, 
+                'parentMessageId': self.parent_message_id, 
+                })
+            
+            # Call Amazon Q
+            response = self.client.chat_sync(**request)
+            self._last_response = response
+
+            # Extract the response text
+            if 'systemMessage' in response:
+                return response["systemMessage"]
+            else:
+                raise ValueError("Unexpected response format from Amazon Q")
+
+        except Exception as e:
+            if "Prompt Length" in str(e):
+                logging.info(f"Prompt Length: {len(prompt)}")
+                print(f"""Prompt:
+                {prompt}""")
+                raise ValueError(f"Error raised by Amazon Q service: {e}")
```

Comment:
Remove this line, as it's redundant with L123. Or change the ValueError message here to specifically reference the prompt length constraint